modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,767 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V2;
+
+use PLL_Language;
+use PLL_Model;
+use stdClass;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Languages REST controller.
+ *
+ * @since 3.7
+ */
+class Languages extends WP_REST_Controller {
+	/**
+	 * @var PLL_Model
+	 */
+	public $model;
```

Comment:
I suggest to have `$languages` instead.
And get rid of `$model->translatable_objects` usage later on.
Instead we could store taxonomies directly in the constructor and use them we need (in only one place actually).
Then the code would be simpler IMHO, only `$this->languages` all along.

What do you think?

Reply:
Good catch. Done in [5bdc9f0](https://github.com/polylang/polylang/pull/1505/commits/5bdc9f0c3157a1ac76c0f84ff6bb3a73392599a3).